carma_cooperative_perception/src/msg_conversion.cpp

```diff
@@ -0,0 +1,158 @@
+  -> carma_cooperative_perception_interfaces::msg::DetectionList
+{
+  carma_cooperative_perception_interfaces::msg::DetectionList detection_list;
+
+  const auto ref_pos_3d{Position3D::from_msg(sdsm.ref_pos)};
+  const Wgs84Coordinate ref_pos_wgs84{
+    ref_pos_3d.latitude, ref_pos_3d.longitude, ref_pos_3d.elevation.value()};
+  const auto ref_pos_utm{project_to_utm(ref_pos_wgs84)};
+
+  for (const auto & object_data : sdsm.objects.detected_object_data) {
+    const auto common_data{object_data.detected_object_common_data};
+
+    carma_cooperative_perception_interfaces::msg::Detection detection;
+    detection.header.frame_id = to_string(ref_pos_utm.utm_zone);
+
+    const auto detection_time{calc_detection_time_stamp(
+      DDateTime::from_msg(sdsm.sdsm_time_stamp),
+      MeasurementTimeOffset::from_msg(common_data.measurement_time))};
+
+    detection.header.stamp = to_time_msg(detection_time);
+
+    detection.id = std::to_string(common_data.detected_id.object_id);
+
+    const auto pos_offset{PositionOffsetXYZ::from_msg(common_data.pos)};
+    const auto utm_displacement{
+      UtmDisplacement{pos_offset.offset_x, pos_offset.offset_y, pos_offset.offset_z.value()}};
+
+    const auto detection_pos_utm{ref_pos_utm + utm_displacement};
+    detection.pose.pose.position = to_position_msg(detection_pos_utm);
+
+    const auto true_heading{units::angle::degree_t{Heading::from_msg(common_data.heading).heading}};
+
+    // Note: This should really use the detection's WGS-84 position, so the
+    // convergence will be off slightly. TODO
+    const auto grid_convergence{calculate_grid_convergence(ref_pos_wgs84, ref_pos_utm.utm_zone)};
+
+    const auto grid_heading{true_heading - grid_convergence};
+    const auto enu_yaw{heading_to_enu_yaw(grid_heading)};
+
+    tf2::Quaternion quat_tf;
+    quat_tf.setRPY(0, 0, remove_units(units::angle::radian_t{enu_yaw}));
+    detection.pose.pose.orientation = tf2::toMsg(quat_tf);
+
+    const auto speed{Speed::from_msg(common_data.speed)};
+    detection.twist.twist.linear.x =
+      remove_units(units::velocity::meters_per_second_t{speed.speed});
+
+    const auto speed_z{Speed::from_msg(common_data.speed_z)};
+    detection.twist.twist.linear.z =
+      remove_units(units::velocity::meters_per_second_t{speed_z.speed});
+
+    const auto accel_set{AccelerationSet4Way::from_msg(common_data.accel_4_way)};
+    detection.accel.accel.linear.x =
+      remove_units(units::acceleration::meters_per_second_squared_t{accel_set.longitudinal});
+    detection.accel.accel.linear.y =
+      remove_units(units::acceleration::meters_per_second_squared_t{accel_set.lateral});
+    detection.accel.accel.linear.z =
+      remove_units(units::acceleration::meters_per_second_squared_t{accel_set.vert});
+
+    detection.twist.twist.angular.z =
+      remove_units(units::angular_velocity::degrees_per_second_t{accel_set.yaw_rate});
+
+    switch (common_data.obj_type.object_type) {
```

Comment:
From sdsm_to_detection_list_node.md it looks like these motion models will be assigned based on configurable parameters. Are these configuration values planned to be captured in future work?

Reply:
Eventually it will be, but let's push this to future work to keep things rolling.